### PR TITLE
feat: csp refactor

### LINKED ISSUE #1154: IPFS: Support Setting Headers
### Overview

When operating from IPFS, the Shapeshift Web v2 application needs the ability to set request/response headers for an assortment of reasons, including but not limited to:
* Security
* Ensuring browsers and other content consumers are able to properly parse content (i.e. `Content-Type`).

### Reference

Direction:
* Use service worker to set headers.
* Service worker architecture will also be required for Pendo integration in the alpha site.
* Only one service worker allowed, so it needs to do everything.

### Acceptance Criteria

Shapeshift Web v2 can properly set request/response headers.

### Screenshots/Mockups

_No response_

### LINKED ISSUE #1899: Generalize CSP generation so Pendo rebases are trivial
### Overview

Currently CSPs have to be added to `headers.js`, and this frequently causes merge conflicts with the [alpha patch](https://github.com/shapeshift/alpha/blob/main/web.patch) that adds the entries Pendo needs.

### References and additional details

All relevant code is currently in `headers.js`.

### Acceptance Criteria

- CSPs are generated by merging exports from all files in a directory
- Each feature can have its CSP additions encapsulated in its own associated file
- Alpha can add its needed Pendo CSP bits by dropping a single file into the relevant directory

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by mrnerdhair**
## Description

Refactors CSP header generation for a better developer experience and ease of composability with add-on CSP rulesets needed for stuff like Pendo. Most CSP headers will be now delivered via meta tag only... which is basically what was happening already since the whole serialized policy is now too long for CF Pages to handle.

Also fixes cypress CSP issues as well.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes #1899. Closes #1154. Closes shapeshift/security#11.
## Risk

This does not change the CSP contents, it just generates it in a different manner. It should be low-risk.

### CODE REVIEW

File: headers.js [hunk 1]
```diff
@@ -1,119 +1,147 @@
...
+    return v.map(entry => {
+      /** @type {[string, string]} */
+      const out = [k, entry]
+      return out
+    })
+  })
+}
+
+/**
+ * @param {CspEntry[]} x
```
Comment (cjthompson) [inline headers.js:22]: Please add a description of the function.

I'd also like to see some tests to verify algorithm.

**Review by cjthompson — DISMISSED** [review]
My only concern is one of documentation and testing. I'd like to see a detailed explanation in a markdown file either here or in the docs repo explaining both the why and the how for future reference.

I'm okay with the documentation being a separate PR.

File: headers.js (hunk 1 above)
Comment (mrnerdhair) [reply headers.js]: Done!

**Review by theoboldfrazier — APPROVED** [review]
Looks good.

**Review by cjthompson — APPROVED** [review]

**Pull request merged**